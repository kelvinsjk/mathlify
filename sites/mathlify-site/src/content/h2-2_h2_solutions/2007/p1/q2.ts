import { Answer } from '$lib/classes/answer';
import { QED } from '$lib/utils/typesetting/utils';
import { Polynomial } from 'mathlify';
import { generateDomain } from '$content/h2-1_h2_learn/01_functions/02_inverse/02_domain/02_practice';
import { improperInverse } from '$content/h2-1_h2_learn/01_functions/02_inverse/03_formula/02_practice';
import { compositeExists } from '$content/h2-1_h2_learn/01_functions/03_composite/01_existence/02_practice';
import { compositeFormula } from '$content/h2-1_h2_learn/01_functions/03_composite/02_formula/02_practice';

export const answer = new Answer();

// ! Complete
// f(x) = 1/(x-3)
// g(x) = x^2

const [a, b, c] = [-3, 0, 1];
const f = new Polynomial([1, a]).reciprocal();
const g = Polynomial.ofDegree(2);
const fState: Parameters<typeof improperInverse>[0] = {
	fnType: 'frac',
	a,
	b,
	c,
	restriction: false,
	unknownConstants: false,
	definition: false
};
const gState = {
	fnType: 'quadratic',
	a: 0,
	b: 0,
	c: 0,
	restriction: false,
	unknownConstants: false
} as const;
const compositeStateFg: Parameters<typeof compositeExists>[0] = {
	f: fState,
	g: gState,
	fg: true
};

// a
{
	const { ans: ans2, soln: soln2 } = compositeExists(compositeStateFg, [f, g]);
	const { ans, soln } = compositeFormula(
		[f, generateDomain(fState), g, generateDomain(gState)],
		false,
		true,
		{ ansInline: true, QED: true }
	);
	answer.addPart(ans + '\\\n' + ans2, soln + '\n\n' + soln2);
}
// b
{
	const { ans, soln } = improperInverse(fState, {
		definition: false,
		rhs: f,
		QED,
		omitReasoningInAns: true
	});
	answer.addPart(ans, soln);
}
