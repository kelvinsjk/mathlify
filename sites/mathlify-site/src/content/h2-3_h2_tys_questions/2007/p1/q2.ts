/* The question here is the copyright of 
- Singapore Examinations and Assessment Board
- Cambridge Assessment International Education
- Ministry of Education, Singapore
- Government of Singapore

We provide this question on Mathlify under fair use for archival purposes.
These are questions more than 10 years old and are difficult to obtain
commercially.

For the newer questions, please purchase a copy of the
"ten year series" from authorized publishers and distributors.
*/

import { Polynomial } from 'mathlify';
import { Question } from '$lib/classes/question';
import { mathlifierDj as mathlifier } from 'mathlifier';

export const question = new Question();

//f(x) = 1/(x-3)
//g(x) = x^2

const a = -3;
const f = new Polynomial([1, a]).reciprocal();
const g = Polynomial.ofDegree(2);

question.addBody(
	mathlifier`Functions ${'f'}
and ${'g'}
are defined by
$${'align*'}
&f: x \\mapsto ${f}  \\quad && \\text{for } x \\in \\mathbb{R}, x \\neq 3, \\\\
&g: x \\mapsto ${g}  \\quad && \\text{for } x \\in \\mathbb{R}.
`
);
// a
{
	question.addPart(
		mathlifier`Only one of the
composite functions ${'fg'}
and ${'gf'}
exists. Give a definition (including the domain)
of the composite that exists, and explain why
the other composite does not exist.`,
		{ marks: 3 }
	);
}
// b
{
	question.addPart(
		mathlifier`Find ${'f^{-1}(x)'}
and state the domain of ${'f^{-1}'}.`,
		{ marks: 3 }
	);
}
