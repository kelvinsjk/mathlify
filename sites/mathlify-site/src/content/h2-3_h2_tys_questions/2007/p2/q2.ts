/* The question here is the copyright of 
- Singapore Examinations and Assessment Board
- Cambridge Assessment International Education
- Ministry of Education, Singapore
- Government of Singapore

We provide this question on Mathlify under fair use for archival purposes.
These are questions more than 10 years old and are difficult to obtain
commercially.

For the newer questions, please purchase a copy of the
"ten year series" from authorized publishers and distributors.
*/

import { Question } from '$lib/classes/question';
import { mathlifierDj as mathlifier } from 'mathlifier';

export const question = new Question();

question.addBody(mathlifier`Out of syllabus (method of differences).`, { marks: 11 });
