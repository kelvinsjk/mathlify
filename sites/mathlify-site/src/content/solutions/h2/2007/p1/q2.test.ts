import { expect, it } from 'vitest';
import { answer } from './q2.soln';

it('2007p1q2', () => {
	expect(answer).toMatchSnapshot();
});
