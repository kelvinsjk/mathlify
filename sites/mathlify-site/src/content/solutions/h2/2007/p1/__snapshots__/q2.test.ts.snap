// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`2007p1q2 1`] = `
Answer {
  "answer": {
    "parts": [
      {
        "body": "<p><math><mrow><mi>g</mi><mi>f</mi><mo lspace="0.2222em" rspace="0.2222em">:</mo></mrow><mrow><mi>x</mi><mo>↦</mo></mrow><mrow><mfrac><mn>1</mn><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mfrac><mo separator="true">,</mo><mspace width="1em"></mspace></mrow><mrow><mo></mo><mrow><mi>x</mi><mo>∈</mo><mi>ℝ</mi><mo separator="true">,</mo><mi>x</mi><mo>≠</mo><mn>3</mn></mrow><mi>.</mi></mrow></math></p>
<p><math><mrow><mi>f</mi><mi>g</mi></mrow></math>
does not exist as <math><mrow><mrow><msub><mi>R</mi><mi>g</mi></msub><mo>⊈</mo><msub><mi>D</mi><mi>f</mi></msub></mrow><mi>.</mi></mrow></math></p>
",
      },
      {
        "body": "<p><math><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo></mrow><mrow><mfrac><mrow><mn>1</mn><mo>+</mo><mn>3</mn><mi>x</mi></mrow><mi>x</mi></mfrac><mi>.</mi></mrow></math>
<br>
<math><mrow><msub><mi>D</mi><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup></msub><mo>=</mo></mrow><mrow><mrow><mo fence="true" form="prefix">(</mo><mo form="prefix" stretchy="false">−</mo><mi>∞</mi><mo separator="true">,</mo><mn>0</mn><mo fence="true" form="postfix">)</mo></mrow><mo>∪</mo></mrow><mrow><mrow><mo fence="true" form="prefix">(</mo><mn>0</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow><mi>.</mi></mrow></math></p>
",
      },
    ],
  },
  "solution": {
    "parts": [
      {
        "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>g</mi><mi>f</mi><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mi>g</mi><mrow><mo fence="true" form="prefix">(</mo><mfrac><mn>1</mn><mrow><mi>x</mi><mo>−</mo><mn>3</mn></mrow></mfrac><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><msup><mrow><mo fence="true" form="prefix">(</mo><mfrac><mn>1</mn><mrow><mi>x</mi><mo>−</mo><mn>3</mn></mrow></mfrac><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mfrac><mn>1</mn><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mfrac></mrow></mtd></mtr></mtable></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mrow><mi>g</mi><mi>f</mi><mo lspace="0.2222em" rspace="0.2222em">:</mo><mi>x</mi><mo>↦</mo><mfrac><mn>1</mn><msup><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mn>2</mn></msup></mfrac><mo separator="true">,</mo><mspace width="1em"></mspace><mi>x</mi><mo>∈</mo><mi>ℝ</mi><mo separator="true">,</mo><mi>x</mi><mo>≠</mo><mn>3</mn><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><msub><mi>R</mi><mi>g</mi></msub></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mrow><mo fence="true" form="prefix">[</mo><mn>0</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><msub><mi>D</mi><mi>f</mi></msub></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mrow><mo fence="true" form="prefix">(</mo><mo form="prefix" stretchy="false">−</mo><mi>∞</mi><mo separator="true">,</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mo>∪</mo><mrow><mo fence="true" form="prefix">(</mo><mn>3</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow></mrow></mtd></mtr></mtable></math></p>
<p>Since <math><mrow><msub><mi>R</mi><mi>g</mi></msub><mo>⊈</mo></mrow><mrow><msub><mi>D</mi><mi>f</mi></msub><mo separator="true">,</mo></mrow></math>
the composite function <math><mrow><mi>f</mi><mi>g</mi></mrow></math>
does not exist <math><mrow><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></math></p>
",
      },
      {
        "body": "<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true"><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mtext>Let </mtext><mi>y</mi><mo>=</mo><mfrac><mn>1</mn><mrow><mi>x</mi><mo>−</mo><mn>3</mn></mrow></mfrac></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>y</mi><mrow><mo fence="true" form="prefix">(</mo><mi>x</mi><mo>−</mo><mn>3</mn><mo fence="true" form="postfix">)</mo></mrow><mo>=</mo><mn>1</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>y</mi><mi>x</mi><mo>−</mo><mn>3</mn><mi>y</mi><mo>=</mo><mn>1</mn></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>y</mi><mi>x</mi><mo>=</mo><mn>1</mn><mo>+</mo><mn>3</mn><mi>y</mi></mrow></mtd></mtr><mtr><mtd style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mi>x</mi><mo>=</mo><mfrac><mrow><mn>1</mn><mo>+</mo><mn>3</mn><mi>y</mi></mrow><mi>y</mi></mfrac></mrow></mtd></mtr></mtable></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mrow><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup><mo form="prefix" stretchy="false">(</mo><mi>x</mi><mo form="postfix" stretchy="false">)</mo><mo>=</mo><mfrac><mrow><mn>1</mn><mo>+</mo><mn>3</mn><mi>x</mi></mrow><mi>x</mi></mfrac><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></math></p>
<p><math display="block" class="tml-display" style="display:block math;"><mtable displaystyle="true" columnalign="right left"><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><msub><mi>D</mi><msup><mi>f</mi><mrow><mo lspace="0em" rspace="0em">−</mo><mn>1</mn></mrow></msup></msub></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><msub><mi>R</mi><mi>f</mi></msub></mrow></mtd></mtr><mtr><mtd class="tml-right" style="padding:0.7ex 0em 0.7ex 0em;"><mrow></mrow></mtd><mtd class="tml-left" style="padding:0.7ex 0em 0.7ex 0em;"><mrow><mo>=</mo><mrow><mo fence="true" form="prefix">(</mo><mo form="prefix" stretchy="false">−</mo><mi>∞</mi><mo separator="true">,</mo><mn>0</mn><mo fence="true" form="postfix">)</mo></mrow><mo>∪</mo><mrow><mo fence="true" form="prefix">(</mo><mn>0</mn><mo separator="true">,</mo><mi>∞</mi><mo fence="true" form="postfix">)</mo></mrow><mspace width="0.2778em"></mspace><mi>∎</mi></mrow></mtd></mtr></mtable></math></p>
",
      },
    ],
  },
}
`;
